docs/migrate-to-nativecurlclient.md

```diff
@@ -0,0 +1,255 @@
+
+If you are using the [Redmine user impersonation](https://www.redmine.org/projects/redmine/wiki/Rest_api#User-Impersonation)
+you have to change your code.
+
+```diff
+// impersonate the user `robin`
+-$client->setImpersonateUser('robin');
++$client->startImpersonateUser('robin');
+
+$userData = $client->getApi('user')->getCurrentUser();
+
+// Now stop impersonation
+-$client->setImpersonateUser(null);
++$client->stopImpersonateUser();
+```
+
+### setCheckSslCertificate() to setCurlOption()
+
+```diff
+-$client->setCheckSslCertificate(true);
++$client->setCurlOption(CURLOPT_SSL_VERIFYPEER, true);
+```
+
+### setCheckSslHost() to setCurlOption()
+
+```diff
+-$client->setCheckSslHost(true);
++$client->setCurlOption(CURLOPT_SSL_VERIFYHOST, 2);
+```
+
+### setSslVersion() to setCurlOption()
+
+```diff
+-$client->setSslVersion(true);
++$client->setCurlOption(CURLOPT_SSLVERSION, CURL_SSLVERSION_DEFAULT);
+```
+
+### setUseHttpAuth() can be removed
+
+Authorization for Redmine uses always the http headers `X-Redmine-API-Key` or
+`Authorization`. So the setting `setUseHttpAuth()` is not longer needed.
+
+```diff
+-$client->setUseHttpAuth(false);
+```
+
+If you want to override the `X-Redmine-API-Key` or `Authorization` header you
+can set them via `setCurlOption()` method.
+
+```diff
+-$client->setUseHttpAuth(false);
++$client->setCurlOption(CURLOPT_HTTPHEADER, ['Authorization: Basic username_password_base64_encoded']);
+// or
++$client->setCurlOption(CURLOPT_HTTPHEADER, ['X-Redmine-API-Key: secret_access_key']);
+```
+
+### setPort() to setCurlOption()
+
+```diff
+-$client->setPort(8080);
++$client->setCurlOption(CURLOPT_PORT, 8080);
+```
+
+### getResponseCode() to getLastResponseStatusCode()
+
+```diff
+-$statusCode = $client->getResponseCode();
++$statuscode = $client->getLastResponseStatusCode();
+```
+
+### setImpersonateUser() to startImpersonateUser()
+
+```diff
+// impersonate the user `robin`
+-$client->setImpersonateUser('robin');
++$client->startImpersonateUser('robin');
+
+$userData = $client->getApi('user')->getCurrentUser();
+
+// Now stop impersonation
+-$client->setImpersonateUser(null);
++$client->stopImpersonateUser();
+```
+
+### setCustomHost() to setCurlOption()
+
+```diff
+-$client->setCustomHost('http://custom.example.com');
++$client->setCurlOption(CURLOPT_HTTPHEADER, ['Host: http://custom.example.com']);
+```
+
+## 2. Stop using deprecated methods
+
+The following methods are deprecated and were set public or protected only for
+testing. They are not available in `NativeCurlClient`. If you are using them,
+please remove them.
+
+- `getUrl()`
+- `decode()`
+- `getCheckSslCertificate()`
+- `getCheckSslHost()`
+- `getSslVersion()`
+- `getUseHttpAuth()`
+- `getPort()`
+- `getImpersonateUser()`
+- `getCustomHost()`
+- `getCurlOptions()`
+- `prepareRequest()`
+- `processCurlResponse()`
+- `runRequest()`
+
+## 3. Switch to `NativeCurlClient`
+
+As the old client the `Redmine\Client\NativeCurlClient` requires:
```

Comment:
```suggestion
Just like with the old `Redmine\Client`, the `Redmine\Client\NativeCurlClient` requires :
```

Reply:
Is having a space before `:` common practice? Personally I havn't seen this before.